app/modules/users/permissions/rules.py

```diff
@@ -269,6 +274,37 @@ def _permitted_via_collaboration(self, user):
         return False
 
 
+# Some modules are special (Submissions) mad may require both access controls in one
+class ModuleOrObjectActionRule(DenyAbortMixin, Rule):
+    def __init__(self, module=None, obj=None, action=AccessOperation.READ, **kwargs):
+        """
+        Args:
+        obj (object) - any object can be passed here, which this functionality will
+            determine whether the current user has enough permissions to write given object
+            object.
+        module (Class) - any class can be passed here, which this functionality will
+            determine whether the current user has enough permissions to perform the action on the class.
+        action (AccessRule) - can be READ, WRITE, DELETE
+        """
+        self._obj = obj
+        self._action = action
+        self._module = module
+        super().__init__(**kwargs)
+
+    def check(self):
+        from app.modules.submissions.models import Submission
+
+        has_permission = False
+        assert self._obj is not None or self._module is not None
+        if self._obj:
+            has_permission = ObjectActionRule(self._obj, self._action).check()
+        else:
+            if self._module == Submission:
+                has_permission = current_user.is_researcher
```

Comment:
I think you said JP that this should be a site admin. 
Is that really the case? Do we want all of this data to be owned by site admins who then have to reassign the ownership for the submission and all the assets in it?

Reply:
When do we trigger the clone from the user perspective? Is it only with out of sync issues? It seems to me that researchers will need this, if not contributors as well but it would happen rarely. Perhaps I'm not thinking of all the situations where we would need to clone.

Reply:
The clone is triggered in the post operation on SubmissionByID so the user would know that it was a separate operation